packages/lesswrong/server/repos/VotesRepo.ts

```diff
@@ -378,6 +378,51 @@ class VotesRepo extends AbstractRepo<"Votes"> {
     return votes
   }
 
+  async getLongtermDownvoteScore(userId: string): Promise<LongtermScoreResult> { // TODO: can't filter for comment count like that! that has to happen elsewhere!
+    return this.getRawDb().one(`
+      SELECT
+        c."userId",
+        ARRAY_AGG(DISTINCT senior_downvoter) AS "longtermSeniorDownvoterIds",
+        COUNT(DISTINCT senior_downvoter) AS "longtermSeniorDownvoterCount",
+        COUNT(c."userId") as "commentCount",
+        (SUM(
+            CASE
+                WHEN
+                    c.total_vote_power < 0
+                    THEN GREATEST((c.total_vote_power * 20)::int, -100)
+                ELSE c.total_vote_power
+            END
+        ) / COUNT(c."userId")) AS "longtermScore"
+      FROM (
+        SELECT
+          full_c.*,
+          SUM(v.power) AS total_vote_power,
+          ARRAY_AGG(DISTINCT v."userId") FILTER (
+            WHERE v.power < 0 AND u.karma > 2000
+          ) AS senior_downvoters
+        FROM "Comments" as full_c
+        LEFT JOIN
+          "Votes" AS v
+          ON full_c._id = v."documentId"
+        LEFT JOIN
+          "Users" AS u ON v."userId" = u._id
+        WHERE
+          v.cancelled = false
+          AND full_c."userId" = $1
+          AND v."userId" != $1
+          AND full_c."postedAt" > CURRENT_DATE - interval '1 year'
+        GROUP BY full_c._id
+      ) AS c
+      LEFT JOIN LATERAL (
+        SELECT senior_downvoter
+        FROM unnest(c.senior_downvoters) AS senior_downvoter
+        WHERE c.total_vote_power < 0
+      ) AS sd ON true
+      GROUP BY c."userId"
+      ORDER BY "longtermScore" DESC
```

Comment:
I haven't reviewed this yet but I see a lateral join which makes me suspect it's too complicated.  (I'm sure there are situations where they're irreplaceable but I literally don't know what they do and have never used one before myself, so I suspect those situations are somewhat rare.)

Reply:
It'd help me review it if you wrote up an english-language explanation of what they query is supposed to be doing, ideally with some pointers in each section